GPlay:   java.lang.NullPointerException - com.owncloud.android.utils.MimeTypeUtil.isImage

Since 2.0.0RC4 (reported for the first time) - Imho not related to RC4 but 2.0.0 in general (search+preview view?)
```
java.lang.RuntimeException: 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2720)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2781)
  at android.app.ActivityThread.-wrap12 (ActivityThread.java)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1508)
  at android.os.Handler.dispatchMessage (Handler.java:102)
  at android.os.Looper.loop (Looper.java:241)
  at android.app.ActivityThread.main (ActivityThread.java:6274)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:886)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:776)
Caused by: java.lang.NullPointerException: 
  at com.owncloud.android.utils.MimeTypeUtil.isImage (MimeTypeUtil.java:257)
  at com.owncloud.android.datamodel.FileDataStorageManager.getVirtualFolderContent (FileDataStorageManager.java:2145)
  at com.owncloud.android.ui.preview.PreviewImagePagerAdapter.<init> (PreviewImagePagerAdapter.java:116)
  at com.owncloud.android.ui.preview.PreviewImageActivity.initViewPager (PreviewImageActivity.java:139)
  at com.owncloud.android.ui.preview.PreviewImageActivity.onAccountSet (PreviewImageActivity.java:482)
  at com.owncloud.android.ui.activity.BaseActivity.onStart (BaseActivity.java:192)
  at com.owncloud.android.ui.activity.DrawerActivity.onStart (DrawerActivity.java:1273)
  at com.owncloud.android.ui.activity.FileActivity.onStart (FileActivity.java:188)
  at com.owncloud.android.ui.preview.PreviewImageActivity.onStart (PreviewImageActivity.java:173)
  at android.app.Instrumentation.callActivityOnStart (Instrumentation.java:1249)
  at android.app.Activity.performStart (Activity.java:6737)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2683)
```

--- a/build.gradle
+++ b/build.gradle
@@ -209,6 +209,8 @@ dependencies {
     annotationProcessor 'org.parceler:parceler:1.1.6'
     implementation 'com.github.bumptech.glide:glide:3.7.0'
     implementation 'com.caverock:androidsvg:1.2.1'
+    implementation "com.android.support:support-annotations:${supportLibraryVersion}"
+
     /// dependencies for local unit tests
     testImplementation 'junit:junit:4.12'
     testImplementation 'org.mockito:mockito-core:1.10.19'
@@ -217,8 +219,7 @@ dependencies {
     androidTestImplementation 'com.android.support.test:rules:1.0.0'
     // Android JUnit Runner
     androidTestImplementation 'com.android.support.test:runner:1.0.0'
-    // Android Annotation Support
-    androidTestImplementation "com.android.support:support-annotations:${supportLibraryVersion}"
+
     // Espresso core
     androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.0'
     // UIAutomator - for cross-app UI tests, and to grant screen is turned on in Espresso tests

--- a/src/main/java/com/owncloud/android/datamodel/FileDataStorageManager.java
+++ b/src/main/java/com/owncloud/android/datamodel/FileDataStorageManager.java
@@ -33,6 +33,7 @@ import android.database.Cursor;
 import android.net.Uri;
 import android.os.RemoteException;
 import android.provider.MediaStore;
+import android.support.annotation.Nullable;
 
 import com.owncloud.android.MainApp;
 import com.owncloud.android.db.ProviderMeta.ProviderTableMeta;
@@ -921,6 +922,7 @@ public class FileDataStorageManager {
         return c;
     }
 
+    @Nullable
     private OCFile createFileInstanceFromVirtual(Cursor c) {
         OCFile file = null;
         if (c != null) {
@@ -2148,16 +2150,18 @@ public class FileDataStorageManager {
         if (c != null && c.moveToFirst()) {
             do {
                 OCFile child = createFileInstanceFromVirtual(c);
-                ocFiles.add(child);
+
+                if (child != null) {
+                    ocFiles.add(child);
+                }
             } while (c.moveToNext());
             c.close();
         }
 
         if (onlyImages) {
-            OCFile current = null;
             Vector<OCFile> temp = new Vector<>();
             for (int i=0; i < ocFiles.size(); i++) {
-                current = ocFiles.get(i);
+                OCFile current = ocFiles.get(i);
                 if (MimeTypeUtil.isImage(current)) {
                     temp.add(current);
                 }

--- a/src/main/java/com/owncloud/android/datamodel/FileDataStorageManager.java
+++ b/src/main/java/com/owncloud/android/datamodel/FileDataStorageManager.java
@@ -2159,9 +2159,10 @@ public class FileDataStorageManager {
         }
 
         if (onlyImages) {
+            OCFile current;
             Vector<OCFile> temp = new Vector<>();
             for (int i=0; i < ocFiles.size(); i++) {
-                OCFile current = ocFiles.get(i);
+                current = ocFiles.get(i);
                 if (MimeTypeUtil.isImage(current)) {
                     temp.add(current);
                 }

